### PR TITLE
NUVIE: Fix passable actor movement

### PR DESCRIPTION
**Pull request by PushmePullyu**
- Do not allow actors marked as passable to pass others who are not

- Allow others to pass actors marked as passable even if they share
    the same movetype (e.g. a mouse should not block the Avatar despite
    both having movetype MOVETYPE_U6_LAND)

- Allow Sherry the mouse to move onto tiles occupied by party members

Fixes [#14396](https://bugs.scummvm.org/ticket/14396)

### CODE REVIEW

File: engines/ultima/nuvie/actors/actor.cpp [hunk 1]
```diff
@@ -422,7 +422,8 @@ bool Actor::can_be_moved() {
 }
 
 bool Actor::can_be_passed(Actor *other) {
-	return (other->is_passable() || is_passable());
+// ethereal actors can always pass us
```
Comment (bluegr) [inline engines/ultima/nuvie/actors/actor.cpp:425]:
```suggestion
	// ethereal actors can always pass us
```

File: engines/ultima/nuvie/actors/u6_actor.cpp [hunk 2]
```diff
@@ -1520,7 +1520,9 @@ bool U6Actor::can_twitch() {
 
 bool U6Actor::can_be_passed(Actor *other) {
 	U6Actor *other_ = static_cast<U6Actor *>(other);
-	return (Actor::can_be_passed(other_) && other_->current_movetype != current_movetype);
+// Sherry the mouse can pass others if they are in a party with her.
```
Comment (bluegr) [inline engines/ultima/nuvie/actors/u6_actor.cpp:1523]:
```suggestion
	// Sherry the mouse can pass others if they are in a party with her.
```

File: engines/ultima/nuvie/actors/actor.cpp [hunk 3]
```diff
@@ -422,7 +422,8 @@ bool Actor::can_be_moved() {
 }
 
 bool Actor::can_be_passed(Actor *other) {
-	return (other->is_passable() || is_passable());
+	// ethereal actors can always pass us
+	return (other->ethereal || is_passable());
```
Comment (bluegr) [inline engines/ultima/nuvie/actors/actor.cpp:426]: You've removed part of the original code here. Shouldn't this be as follows?

```suggestion
	return (other->is_passable() || is_passable() ||| other->ethereal);
```

Comment (PushmePullyu) [reply engines/ultima/nuvie/actors/actor.cpp:426]: Thanks for the review. The removal is intentional: An actor marked as passable should not automatically be able to pass others. The "ethereal" check was done in "other->is_passable()" before and is still needed.

**Comment by sev-** [comment]
Thanks!

**Pull request merged**